server/protocols/webadmin-integration-test/src/test/java/org/apache/james/webadmin/integration/EventDeadLettersIntegrationTest.java

```diff
@@ -0,0 +1,464 @@
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.webadmin.integration;
+
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.when;
+import static io.restassured.RestAssured.with;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.james.CassandraJmapTestRule;
+import org.apache.james.DockerCassandraRule;
+import org.apache.james.GuiceJamesServer;
+import org.apache.james.event.json.EventSerializer;
+import org.apache.james.mailbox.DefaultMailboxes;
+import org.apache.james.mailbox.events.Event;
+import org.apache.james.mailbox.events.Group;
+import org.apache.james.mailbox.events.MailboxListener;
+import org.apache.james.mailbox.inmemory.InMemoryId;
+import org.apache.james.mailbox.inmemory.InMemoryMessageId;
+import org.apache.james.mailbox.probe.MailboxProbe;
+import org.apache.james.modules.MailboxProbeImpl;
+import org.apache.james.probe.DataProbe;
+import org.apache.james.utils.DataProbeImpl;
+import org.apache.james.utils.WebAdminGuiceProbe;
+import org.apache.james.webadmin.WebAdminUtils;
+import org.apache.james.webadmin.routes.EventDeadLettersRoutes;
+import org.apache.james.webadmin.routes.TasksRoutes;
+import org.eclipse.jetty.http.HttpStatus;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import com.google.inject.multibindings.Multibinder;
+
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+
+public class EventDeadLettersIntegrationTest {
+
+    public static class RetryEventsListenerGroup extends Group {}
+
+    public class RetryEventsListener implements MailboxListener.GroupMailboxListener {
+        final Group GROUP = new RetryEventsListenerGroup();
+
+        private int retriesBeforeSuccess;
+        private Map<Event.EventId, Integer> retries;
+        private List<Event> failedEvents;
+        private List<Event> successfulEvents;
+
+        private void init() {
+            this.retriesBeforeSuccess = 0;
+            this.retries = new HashMap<>();
+            this.failedEvents = new ArrayList<>();
+            this.successfulEvents = new ArrayList<>();
+        }
+
+        RetryEventsListener() {
+            init();
+        }
+
+        @Override
+        public Group getDefaultGroup() {
+            return GROUP;
+        }
+
+        @Override
+        public void event(Event event) throws Exception {
+            int currentRetries = 0;
+            if (retries.containsKey(event.getEventId())) {
+                currentRetries = retries.get(event.getEventId());
```

Comment:
Realocation spotted.

Can we try to rewrite this with an Optional? (and maybe do a method extract)?